app/services/image_filename_service.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+class ImageFilenameService
+  MAX_LENGTH = 65
+
+  def initialize(revision)
+    @revision = revision
+  end
+
+  def call(suggested_name, mime_type)
+    filename = ActiveStorage::Filename.new(suggested_name)
+
+    base = filename.base.parameterize.slice 0...MAX_LENGTH
+    base = ensure_unique(base)
+
+    extension = Rack::Mime::MIME_TYPES.invert[mime_type]
```

Comment:
I think doing this here could have some unexpected consequences. Impact is likely low with images but could be worse for attachments.

First off the `Rack::Mime::MIME_TYPES` is a different database of mime types to what ActiveStorage uses which comes from [MimeMagic](https://github.com/minad/mimemagic)

As invert also just uses the first thing we could end up with some unlikely extensions:

E.g:

```
irb(main):019:0> Rack::Mime::MIME_TYPES[".svg"]
=> "image/svg+xml"
irb(main):018:0> Rack::Mime::MIME_TYPES.invert["image/svg+xml"]
=> ".svgz"
```

Frustratingly mime magic doesn't seem to have an aspect of the most common or preferred extension in it's tables which gives you:

```
irb(main):026:0> MimeMagic.new("image/jpeg").extensions
=> ["jpe", "jpeg", "jpg"]
```

As I feel like I'm opening up a rabbit hole here it might be worth having a quick chat to see what's simplest route forward.

Reply:
Sure - whenever you're free 👍 .

Reply:
Updated to just preserve any extension that comes with the uploaded file.